moshi/src/test/java/com/squareup/moshi/EnumJsonAdapterTest.java

```diff
@@ -0,0 +1,73 @@
+package com.squareup.moshi;
+
+import java.io.IOException;
+import okio.Buffer;
+import org.junit.Test;
+
+import static com.squareup.moshi.Util.NO_ANNOTATIONS;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class EnumJsonAdapterTest {
+  private final Moshi moshi = new Moshi.Builder().build();
+
+  enum Roshambo {
+    ROCK,
+    PAPER,
+    SCISSORS
+  }
+
+  @Test public void basics() throws Exception {
+    JsonAdapter<Roshambo> adapter = moshi.adapter(Roshambo.class).lenient();
+    assertThat(adapter.fromJson("\"ROCK\"")).isEqualTo(Roshambo.ROCK);
+    assertThat(adapter.toJson(Roshambo.PAPER)).isEqualTo("\"PAPER\"");
+  }
+
+  @Test public void invalidInput() throws Exception {
+    JsonAdapter<Roshambo> adapter = moshi.adapter(Roshambo.class).lenient();
+    assertThat(adapter.fromJson("\"Lizard\"")).isNull();
+    assertThat(adapter.fromJson("\"SPOCK\"")).isNull();
+  }
+
+  @Test public void nulls() throws Exception {
+    JsonAdapter<Roshambo> adapter = moshi.adapter(Roshambo.class).lenient();
+    assertThat(adapter.fromJson("null")).isNull();
+    assertThat(adapter.toJson(null)).isEqualTo("null");
+  }
+
+  static class Game {
+    Roshambo player1;
+    Roshambo player2;
+  }
+
+  @Test public void enumInObject() throws Exception {
```

Comment:
I don't think this test is necessary; we already test composition in other tests.